tmt/steps/provision/__init__.py

```diff
@@ -464,25 +463,32 @@ def stop(self):
 
         raise NotImplementedError()
 
-    def reboot(self, hard=False):
+    def reboot(self, hard=False, command=None, timeout=None):
         """
         Reboot the guest, return True if successful
 
         Parameter 'hard' set to True means that guest should be
         rebooted by way which is not clean in sense that data can be
         lost. When set to False reboot should be done gracefully.
+
+        Use the 'command' parameter to specify a custom reboot command
+        instead of the default 'reboot'.
+
+        Parameter 'timeout' can be used to specify time (in seconds) to
+        wait for the guest to come back up after rebooting.
         """
 
         raise NotImplementedError()
 
-    def reconnect(self, timeout=CONNECTION_TIMEOUT):
+    def reconnect(self, timeout=None):
         """
         Ensure the connection to the guest is working after reboot
 
         The default timeout is 1h to allow long operations such as
```

Comment:
The information WRT the default timeout is no longer true. Maybe a f-string & `{CONNECTION_TIMEOUT / 60}` could prevent any future inconsistencies.